shopfloor_single_product_transfer/services/single_product_transfer.py

```diff
@@ -0,0 +1,588 @@
+        # TODO qty done < product_qty: transfer_confirm_done
+        picking = move_line.picking_id
+        picking._action_done()
+        message = self.msg_store.transfer_done_success(picking)
+        return self._response_for_select_location(message=message)
+
+    def _start__find_ongoing_moves_for_user_domain(self, user):
+        return [
+            ("shopfloor_user_id", "=", user.id),
+            ("picking_id.picking_type_id", "in", self.picking_types.ids),
+            ("state", "in", ("assigned", "partially_available")),
+        ]
+
+    def _start__find_ongoing_moves_for_user(self):
+        user = self.env.user
+        domain = self._start__find_ongoing_moves_for_user_domain(user)
+        return self.env["stock.move.line"].search(domain, limit=1)
+
+    def _set_quantity__by_product(self, move_line, barcode, confirmation=False):
+        product_handlers = [
+            self._set_quantity__scan_product,
+            self._set_quantity__scan_lot,
+            # scan packaging
+        ]
+        product_response = self._use_handlers(product_handlers, move_line, barcode)
+        if product_response:
+            return product_response
+
+    def _set_quantity__by_location(self, move_line, barcode, confirmation=False):
+        search = self._actions_for("search")
+        location = search.location_from_scan(barcode)
+        handlers = [
+            self._set_quantity__check_location,
+            self._set_quantity__post_move,
+        ]
+        if location:
+            response = self._use_handlers(
+                handlers, location, move_line, confirmation=confirmation
+            )
+            if response:
+                return response
+
+    # Endpoints
+
+    def start(self):
+        ongoing_move_line = self._start__find_ongoing_moves_for_user()
+        if ongoing_move_line:
+            message = self.msg_store.recovered_previous_session()
+            return self._response_for_set_quantity(ongoing_move_line, message=message)
+        return self._response_for_select_location()
+
+    def scan_location(self, barcode):
+        search = self._actions_for("search")
+        location = search.location_from_scan(barcode)
+        handlers = [
+            self._scan_location__location_found,
+            self._scan_location__check_location,
+            self._scan_location__check_stock,
+        ]
+        response = self._use_handlers(handlers, location)
+        if response:
+            return response
+        return self._response_for_select_product(location)
+
+    @with_savepoint
+    def scan_product(self, location_id, barcode):
+        """Looks for a move line in the given location, from a barcode."""
+        location = self.env["stock.location"].browse(location_id)
+        if not location.exists():
+            return self._response_for_select_product(location)
+        handlers = [
+            self._scan_product__scan_product,
+            self._scan_product__scan_lot,
+        ]
```

Comment:
missing `_scan_product__scan_packaging` for scanning a product packaging

Reply:
done